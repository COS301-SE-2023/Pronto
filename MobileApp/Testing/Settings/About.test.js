import React from 'react';
import { render } from '@testing-library/react-native';
import AboutScreen from '../../screens/Settings/About';
import expect from "expect";

describe('AboutScreen', () => {
    it('renders correctly', () => {
        const { getByText,getByTestId } = render(<AboutScreen />);

        const heading = getByText('About Pronto');
        const text1 = getByTestId('intro');
        expect(text1).toBeTruthy();

        const text2 = getByTestId('mission');
        expect(text2).toBeTruthy();

        const keyFeatures1 = getByTestId('keyFeatures1');
        expect(keyFeatures1).toBeTruthy();

        const keyFeatures2 = getByTestId('keyFeatures2');
        expect(keyFeatures2).toBeTruthy();

        const keyFeatures3 = getByTestId('keyFeatures3');
        expect(keyFeatures3).toBeTruthy();

        const keyFeatures4 = getByTestId('keyFeatures4');
        expect(keyFeatures4).toBeTruthy();

        const keyFeatures5 = getByTestId('keyFeatures5');
        expect(keyFeatures5).toBeTruthy();

        const message = getByTestId('message');
        expect(message).toBeTruthy();


        const subheading = getByText('Key Features:');
        const listItem1 = getByText('1. Schedule Management: Load and track your lecture schedules, assignment due dates, test schedules, and examination dates all in one place.');
        const listItem2 = getByText('2. Reminders and Notifications: Set up personalized reminders for upcoming tasks and receive notifications to ensure you stay on top of your academic responsibilities.');

        expect(heading).toBeTruthy();
        expect(subheading).toBeTruthy();
        expect(listItem1).toBeTruthy();
        expect(listItem2).toBeTruthy();

    });


    it('matches snapshot', () => {
        const { toJSON } = render(<AboutScreen />);
        expect(toJSON()).toMatchSnapshot();
    });
});
