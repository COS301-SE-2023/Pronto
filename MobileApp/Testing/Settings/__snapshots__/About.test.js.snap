// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutScreen matches snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "padding": 20,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 10,
        }
      }
    >
      About Pronto
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 10,
        }
      }
      testID="intro"
    >
      Pronto is a web/mobile application designed specifically for students to efficiently manage their academic tasks, schedules, and deadlines. With Pronto, you can stay organized and never miss an important assignment or exam again.
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 10,
        }
      }
      testID="mission"
    >
      Our mission is to provide a single, consolidated platform for students and educational institutions to enhance the learning experience. We believe that by leveraging technology, students can optimize their time and focus on what truly matters - their education.
    </Text>
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 10,
          "marginTop": 20,
        }
      }
    >
      Key Features:
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
      testID="keyFeatures1"
    >
      1. Schedule Management: Load and track your lecture schedules, assignment due dates, test schedules, and examination dates all in one place.
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
      testID="keyFeatures2"
    >
      2. Reminders and Notifications: Set up personalized reminders for upcoming tasks and receive notifications to ensure you stay on top of your academic responsibilities.
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
      testID="keyFeatures3"
    >
      3. Campus Maps: Access campus maps to easily navigate your educational institution's facilities and find your way around.
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
      testID="keyFeatures4"
    >
      4. Institution Pages: Educational institutions can maintain their own page within the app to provide students with important information, such as class schedules, campus events, and announcements.
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
      testID="keyFeatures5"
    >
      5. User-Friendly Interface: Pronto offers an intuitive and user-friendly interface, making it easy for students of all levels to navigate and utilize the app effectively.
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 10,
        }
      }
      testID="message"
    >
      We are committed to continuously improving and expanding our services to meet the needs of students and educational institutions. Your feedback is valuable to us, and we encourage you to reach out to us with any suggestions or concerns you may have.
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 10,
        }
      }
    >
      Thank you for choosing Pronto as your academic companion!
    </Text>
  </View>
</RCTScrollView>
`;
